polynote-frontend/polynote/cell.js

```diff
@@ -352,21 +353,25 @@ export class CodeCell extends Cell {
         this.updateKeyHandler(KeyPress.of("Enter", {shift: true}).h, (prev) => {
             return () => {
                 this.dispatchEvent(new RunCellEvent(this.id));
-                prev();
+                return prev();
             }
         });
         this.updateKeyHandler(KeyPress.of("Enter", {shift: true, meta: true}).h, (prev) => {
             return () => {
                 this.dispatchEvent(new RunCellEvent(this.id));
-                prev();
+                return prev();
             }
         });
 
         // actually bind keydown
         this.editor.onKeyDown((evt) => {
-            if (this.onKeyDown(evt)) {
-                evt.stopPropagation();
-                evt.preventDefault();
+            // this is really ugly, is there a better way to tell whether the widget is visible??
+            const suggestionsVisible = this.editor.getContribution('editor.contrib.suggestController')._widget._value.suggestWidgetVisible.get();
+            if (!suggestionsVisible) { // don't do stuff when suggestions are visible
```

Comment:
Earlier if you got suggestions on the last spot in a cell (e.g., imagine your cell is `spark.` and you just pressed `.`), pressing the down arrow would close the suggestions and move you to the next cell, rather than allowing you to select a suggestion. This way we know whether the suggestions are visible (through this hack) and refrain from nastily stealing focus away from them.